File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int n=tokens.size();
        stack<int>st;
        int i=0;
        while(i<n)
        {
            if(tokens[i]=="*" || tokens[i]=="+" || tokens[i]=="-" ||
            tokens[i]=="/")
            {
                if(st.size()>=2)
                {
                    int top2=st.top();
                    st.pop();
                    int top1=st.top();
                    st.pop();
                    if(tokens[i]=="*")
                    st.push(top1*top2);
                    else if(tokens[i]=="+")
                    st.push(top1+top2);
                    else if(tokens[i]=="-")
                    st.push(top1-top2);
                    else
                    st.push(top1/top2);
                }
            }
            else
            {
                st.push(stoi(tokens[i]));
            }
            i++;
        }
        if(!st.empty())
        return st.top();
        else
        return -1;
        
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-4.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st=[]
        n=len(temperatures)
        ans=[0]*n
        i=n-1
        while(i>=0):
            while(len(st)>0 and  temperatures[i]>=temperatures[st[-1]]):
                st.pop()
            if(len(st)>0):
                ans[i]=st[-1]-i
            st.append(i)
            i-=1
        return ans
        
                

        